conda/plugins/subcommands/doctor/health_checks.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+from __future__ import annotations
+
+import json
+from pathlib import Path
+
+OK_MARK = "✅"
+REPORT_TITLE = "\nENVIRONMENT HEALTH REPORT\n"
+DETAILED_REPORT_TITLE = "\nDETAILED ENVIRONMENT HEALTH REPORT\n"
+MISSING_FILES_SUCCESS_MESSAGE = f"{OK_MARK} There are no packages with missing files.\n"
+
+
+def display_report_heading(prefix: str) -> None:
+    """
+    Displays our report heading
+    """
+    environment = Path(prefix)
+    environment_name = environment.name
+    print("-" * 20)
+    print(REPORT_TITLE)
+    print(f"Environment Name: {environment_name}\n")
+
+
+def get_number_of_missing_files(prefix: str) -> dict[str, int]:
+    """
+    Returns a dictionary with packages and the number of missing files in them
+    """
+    packages_with_missing_files = find_packages_with_missing_files(prefix)
+
+    return {k: len(v) for k, v in packages_with_missing_files.items()}
+
+
+def find_packages_with_missing_files(prefix: str | Path) -> dict[str, list[str]]:
+    """
+    Finds packages listed in conda-meta which have missing files
+    """
+    packages = {}
+    prefix = Path(prefix)
+    conda_meta = prefix.joinpath("conda-meta")
+    for file in conda_meta.iterdir():
+        if file.name.endswith(".json"):
+            name = str(file.name)[:-5]
+            packages[name] = []
+            with file.open() as f:
+                data = json.load(f)
```

Comment:
Let's use the more concise operation provided by `pathlib.Path`:

```suggestion
            data = json.loads(file.read_text())
```